website/server/controllers/api-v3/groups.js

```diff
@@ -402,6 +409,11 @@ api.leaveGroup = {
     }
 
     await group.leave(user, req.query.keep);
+
+    if (user.newMessages[group._id]) {
```

Comment:
my thinking was that the function removed messages associated with a specific group_id, the if check remaining where it was allows the function to be more generic, and the logic for validating if it's needed is kept at that call site.   prefer i flip the location?